### PR TITLE
Refactor rfid processing

### PR DESCRIPTION
**Pull request by matjack1**
Please merge first https://github.com/esprfid/esp-rfid/pull/490 as this will be easier to review.

Unfortunately this is changing heavily the rfid.esp file. So I understand if you are skeptical about merging this, but this refactor was needed to fix a few bugs, add better pincode support and make it more reliable.

I'm trying to explain what happens now, in particular with the wiegand sensor with pincode.

We have now two state machines, one for the sensor state, another for the processing state.
Everything starts at the `rfidLoop()`, where now the steps are more separate. The `rfidRead()` takes care of getting the value of the sensor and setting the state in `rfidState`. Then, if the pincode is required the following step is going into the `pinCodeRead` function, which adds the pincode.

Once we have the info in memory and the reading steps are done (tracked in the `rfidState`), we can proceed with the `rfidProcess()` function. There we check if the card/pincode combination is valid and if the card has necessary permissions. If so we change the `processingState` variable (the other "state machine") so that the next step can proceed and we trigger the relays if necessary.

In the last step `rfidOutsideMessaging()` we send the necessary MQTT messages and signal how things are going with LEDs, so that also the user can immediately see what's going on.

I'm happy to go over this code over a call if you want!

I have a similar important refactor for the WiFi that is coming soon, so that we fix a few outstanding issues there. But this needs to go first to avoid too many changes at once (if these are not enough already!).

### CODE REVIEW

**Review by gdalfovo — APPROVED** [review]

File: src/config.h [hunk 1]
```diff
@@ -7,10 +7,12 @@ struct Config {
...
     unsigned long activateTime[MAX_NUM_RELAYS];
     unsigned long autoRestartIntervalSeconds = 0;
+    uint8_t beeperpin = 255;
     char *deviceHostname = NULL;
     uint8_t doorbellpin = 255;
     uint8_t doorstatpin = 255;
     char *httpPass = NULL;
+    uint8_t ledwaitingpin = 255;
     int lockType[MAX_NUM_RELAYS];
     uint8_t maxOpenDoorTime = 0;
```
Comment (gdalfovo) [inline src/config.h]: Change the type of maxOpenDoorTime to: unsigned long

**Review by gdalfovo — CHANGES_REQUESTED** [review]

File: src/config.h (hunk 1 above)
Comment (matjack1) [reply src/config.h]: Thank you, fixed!

**Review by omersiar — APPROVED** [review]
Looks good to me

**Comment by omersiar** [comment]
Thank you guys. much appreciated.

**Pull request merged**